### ISSUE TITLE
Migrate off tsdx

### ISSUE THREAD

**Issue by karlhorky** [upleveled/preflight#126]
`tsdx` was a nice idea, but it's [no longer maintained](https://github.com/jaredpalmer/tsdx/issues/1065).

Time to come up with our own solution for this, maybe using:

- `esbuild`
- [`tsm`](https://github.com/lukeed/tsm)
- [`tsx`](https://github.com/privatenumber/tsx)

**Comment by karlhorky** [comment]
Wonder if this starter could offer us any tips: https://github.com/egoist/ts-lib-starter

**Comment by karlhorky** [comment]
Or, also consider moving to one of:

- `microbundle`
- `unbuild`
- `preconstruct`

**Comment by karlhorky** [comment]
Maybe just use `tsc` alone: https://cmdcolin.github.io/posts/2022-05-27-youmaynotneedabundler

**Comment by karlhorky** [comment]
Maybe another example, this one using `esbuild`: https://github.com/mmazzarolo/tangerine-monorepo

**Comment by karlhorky** [comment]
Examples of switching to `tsup`:

- https://github.com/immerjs/immer/pull/1032
- https://github.com/byCedric/expo-monorepo-example/pull/46

**Comment by karlhorky** [comment]
Another reason to switch:

During the switch from Yarn to pnpm (#379 ), we also had to add some additional weird configuration:

`.eslintignore`

```diff
dist
babel.config.cjs
prettier.config.cjs
+__tests__
+bin
```

`.github/workflows/build-lint-test.yml`

```yaml
      - name: Check types
        run: |
          pnpm tsc --project __tests__/tsconfig.json
          pnpm tsc --project bin/tsconfig.json
          pnpm tsc --project tsconfig.config.json
```

<img width="234" alt="Screenshot 2023-04-14 at 10 27 31" src="https://user-images.githubusercontent.com/1935696/231989352-bba4911c-8bab-4f1b-a2f9-0ca6c38ada5f.png">

**Comment by ProchaLu** [comment]
My reason for switching:
After the issues encountered with TSDX and the switch to pnpm in the PR https://github.com/upleveled/preflight/pull/379, we have to consider migrating away from TSDX. Without switching, we have to use `node-linker=hoisted`, as a temporary workaround, but this is not a good solution for the future.

Different solutions without switching what I tried where to use `rollup-plugin-typescript2` package to version 0.32.0 in my `package.json` file using PNPM's overrides feature. 
```
  "pnpm": {
    "overrides": {
      "rollup-plugin-typescript2": "^0.32.0"
    }
  }
```
However, when I tried this approach, I encountered a semantic error related to conflicting versions of dependencies. I also explored other workarounds, such as using shamefully-hoist=true with PNPM, but that resulted in another error related to untyped function calls not accepting type arguments.
